geode-core/src/main/java/org/apache/geode/internal/security/shiro/JMXShiroAuthenticator.java

```diff
@@ -76,8 +79,17 @@ public void handleNotification(Notification notification, Object handback) {
     if (notification instanceof JMXConnectionNotification) {
       JMXConnectionNotification cxNotification = (JMXConnectionNotification) notification;
       String type = cxNotification.getType();
-      if (JMXConnectionNotification.CLOSED.equals(type)) {
-        this.securityService.logout();
+      String connectionId = cxNotification.getConnectionId();
+      synchronized (connectedUsers) {
```

Comment:
With changing the containsKey+put to putIfAbsent, I think you can change the map to Collections.synchronizedMap or a ConcurrentHashMap and remove this synchronized block.